### PR TITLE
.Net: Extract semantic functions from Kernel.Core

### LINKED ISSUE #2563: .Net: Extract semantic functions from Kernel.Core
Semantic functions will be moved to core skills project

```[tasklist]
### Tasks
- [ ] Move code
- [ ] Create an ADR to describe the change, project naming and impact for consumers
- [ ] Make required build/pipeline changes
```

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

  1. Why is this change required? This is part of the work to extract functionality from Semantic Kernel core prior to 1.0 release.
  2. What problem does it solve? Isolates the semantic function implementation from the Semantic Kernel core.
  3. What scenario does it contribute to? Contributes to the goal of reducing the Semantic Kernel core to focus on AI orchestration.
  4. If it fixes an open issue, please link to the issue here. https://github.com/microsoft/semantic-kernel/issues/2563
Fixes #2563
### Description

- Move classes from abstractions SemanticFunctions folder to new functions.semantic project
- Move classes from core SemanticFunctions and SkillDefinition folders to new functions.semantic project
- Fix up all unit tests

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Extensions/Planning.ActionPlanner/Planning.ActionPlanner.csproj
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Planning.ActionPlanner/Planning.ActionPlanner.csproj:1]: Whoops, bad merge I think. You don't want these anymore.

File: dotnet/src/Functions/Functions.Semantic/Extensions/KernelSemanticFunctionExtensions.cs [hunk 1]
```diff
@@ -22,6 +23,43 @@ namespace Microsoft.SemanticKernel;
...
+    /// <param name="kernel">Semantic Kernel instance</param>
+    /// <param name="functionName">Name of the semantic function. The name can contain only alphanumeric chars + underscore.</param>
+    /// <param name="functionConfig">Function configuration, e.g. I/O params, AI settings, localization details, etc.</param>
+    /// <returns>A C# function wrapping AI logic, usually defined with natural language</returns>
+    public static ISKFunction RegisterSemanticFunction(
+        this IKernel kernel,
+        string functionName,
+        SemanticFunctionConfig functionConfig)
+    {
+        return kernel.RegisterSemanticFunction("TODO", functionName, functionConfig);
```
Comment (SergeyMenshykh) [inline dotnet/src/Functions/Functions.Semantic/Extensions/KernelSemanticFunctionExtensions.cs:38]: Is "TODO" expected as the plugin name?

**Review by lemillermicrosoft — APPROVED** [review]
Remove old csproj files, otherwise LGTM.

File: dotnet/src/Functions/Functions.Semantic/Extensions/KernelSemanticFunctionExtensions.cs (hunk 1 above)
Comment (markwallace-microsoft) [reply dotnet/src/Functions/Functions.Semantic/Extensions/KernelSemanticFunctionExtensions.cs]: I've made `FunctionsCollection.GlobalFunctionsPluginName` public and we use that everywhere.

**Review by SergeyMenshykh — APPROVED** [review]

**Pull request merged**